src/nspawn/nspawn.c

```diff
@@ -2796,6 +2827,10 @@ static int outer_child(
         if (r < 0)
                 return r;
 
+        r = setup_machine_id(directory);
```

Comment:
Yeah, @keszybz has a point, instead of first writing the file, and then reading it back, let's just keep it around internally...